pageserver/src/layered_repository.rs

```diff
@@ -1366,26 +1381,25 @@ impl LayeredTimeline {
                 ancestor_timeline: ancestor_timelineid,
                 ancestor_lsn: self.ancestor_lsn,
             };
+
             LayeredRepository::save_metadata(
                 self.conf,
                 self.timelineid,
                 self.tenantid,
                 &metadata,
                 false,
             )?;
+            if self.upload_relishes {
+                schedule_timeline_upload(
```

Comment:
- Registering SyncTask after each checkpoint is far from optimal. Pageserver performs loads of checkpoints, primarily to be able to free the memory and most of those layers will be removed by GC in the nearest future, so uploading them to S3 is a waste of resources...
So I propose having a separate schedule for the uploader, just like a backup schedule for a regular postgres database.

Reply:
Yes, that was really noisy before https://github.com/zenithdb/zenith/pull/764#discussion_r730384130 indeed.
But after that PR merge, we should already upload things only if `disk_consistent_lsn` changes which is exactly what we want, I think.

Or do you think the thread is more beneficial still?